src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+        for (Task task : tasks) {
+            if (task == null) {
+                break;
+            }
+            System.out.println(task);
+        }
+        System.out.print(dashes);
+    }
+
+    // add task: when user types in a task (not list, done or bye command).
+    // @params in: Task instance of new task created + string of task name
+    public static void addTask(Task newTask, String taskName) {
+        tasks[tasksCounter++] = newTask;
+        System.out.print(dashes);
+        System.out.println("Very well, adding task \"" + taskName + "\"");
+        System.out.print(dashes);
+
+    }
+
+    public static void addToDo(String taskName) {
+        Todo newTodo = new Todo(false, taskName);
+        addTask(newTodo, taskName);
+    }
+
+    public static void addDeadline(String taskName, String deadline) {
+        Deadline newDeadline = new Deadline(false, taskName, deadline);
+        addTask(newDeadline, taskName);
+    }
+
+    public static void addEvent(String taskName, String date) {
+        Event newEvent = new Event(false, taskName, date);
+        addTask(newEvent, taskName);
+    }
+
+    private static void mainProgram() {
+        //String userInput = "";
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            String userInput = in.nextLine();
+            while (userInput.trim().isEmpty()) {
+                userInput = in.nextLine();
+            }
+
+            if (userInput.equals("bye")) {
+                break;
+            }
+            // if user lists
+            if (userInput.equals("list")) {
+                getList();
+                continue;
+            }
+
+            // TODO
+            if (userInput.startsWith("todo")) {
+                addToDo(userInput.substring(userInput.indexOf(" ") + 1));
+                continue;
+            }
+
+            // DEADLINES
+            if (userInput.startsWith("deadline")) {
+                String taskName = userInput.substring(userInput.indexOf(" ") + 1, userInput.indexOf("/")).trim();
+                String deadline = userInput.substring(userInput.indexOf("/") + 3).trim();
+                addDeadline(taskName, deadline);
+                continue;
+            }
+
+            //EVENTS
+            if (userInput.startsWith("event")) {
+                String taskName = userInput.substring(userInput.indexOf(" ") + 1, userInput.indexOf("/")).trim();
+                String eventDate = userInput.substring(userInput.indexOf("/") + 3).trim();
+                addEvent(taskName, eventDate);
+                continue;
+            }
+
+            // if user wants to mark as done
+            if (userInput.startsWith("done")) {
+                // get index of task to change
+                int taskIndex = Integer.parseInt(userInput.substring(userInput.indexOf(" ") + 1)) - 1;
+
+                // if task out of bounds
+                if (taskIndex < 0 || taskIndex > 99) {
```

Comment:
Can create a boolean value for out of bounds index so that it is more easily understood.